### PR TITLE
Deferred comment indicator improvements

### PR DESCRIPTION
**Pull request by micahmo**
## Pull Request Description

This PR adds a few QoL improvements to the deferred comment indicator.

1. Localize and handle pluralization.
2. Lighten color so it's not mistaken for a comment.
3. Add InkWell effect.
4. Better handling of errors.
   - Show toast for errors.
   - Don't show spinner forever.

## Issue Being Fixed

Issue Number: N/A

## Screenshots / Recordings

https://github.com/thunder-app/thunder/assets/7417301/c6b281e5-c59f-4fe4-856b-e48693213c66

https://github.com/thunder-app/thunder/assets/7417301/41d92504-0dfc-4e9b-9315-622a96926209

## Checklist

- [x] Did you update CHANGELOG.md?
- [x] Did you use localized strings where applicable?
- [ ] Did you add `semanticLabel`s where applicable for accessibility?

### CODE REVIEW

File: lib/post/pages/post_page.dart
Comment (ajsosa) [inline lib/post/pages/post_page.dart:1]: Are we moving away from floating toasts? If that's the case, we should probably make them `fixed` across the board so it's consistent. Thoughts?

Comment (micahmo) [reply lib/post/pages/post_page.dart:1]: > Are we moving away from floating toasts?

Yes I think so. In this case, it didn't appear until I removed floating. Also the non-floating ones automatically bump the FAB.

> If that's the case, we should probably make them `fixed` across the board so it's consistent. Thoughts?

I agree. @hjiangsu suggested we do an overall snackbar refactor so we don't have these same lines duplicated everywhere. Then we could have one place that controls the style.

Comment (hjiangsu) [reply lib/post/pages/post_page.dart:1]: Yeah, I think we can do a refactor of the snackbar stuff in another PR to consolidate the codebase and to keep things consistent for all toast notifications!

**Review by hjiangsu — APPROVED** [review]
I think it looks good!

**Comment by micahmo** [comment]
Resolved conflicts!

**Pull request merged**